### PR TITLE
conduit: add Fujitsu fortran options

### PR DESCRIPTION
**Pull request by ketsubouchi**
1. Cmake version
`cmake@3.8.2:3.17.9999` is not equal to `# cmake 3.8.2 or newer`
So I fixed version specification.

2. Fujitsu Fortran options
Cmake cannot recognize Fujitsu Fortran compiler now.
So set Fortran compiler options manually. 
Patch for Cmake is working now, so this patch is tentative.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@cyrush This seems good to go as far as I can see, but waiting for your green light before merging.

File: var/spack/repos/builtin/packages/conduit/package.py
Comment (cyrush) [inline var/spack/repos/builtin/packages/conduit/package.py:89]: the 3.17 constraint was added b/c of a recent issue with CMake 3.18.0 and CMake's FindHDF5 logic:
https://github.com/spack/spack/pull/17937

However -- now that spack has cmake 3.18.1 (and 3.18.2), which I belive have a fix for this issue -- I think its safe to again relax this constraint.

**Review by cyrush — APPROVED** [review]
Looks good to me! I added a comment about the cmake constraint, which I think we can now remove.

**Pull request merged**
